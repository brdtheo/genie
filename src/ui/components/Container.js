import React from "react";

export default function Container(props) {
  return <div className="w-[56rem] m-auto py-6">{props.children}</div>;
}
