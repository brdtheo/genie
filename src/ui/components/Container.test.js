/**
 * @jest-environment jsdom
 */
import * as React from "react";
import renderer from "react-test-renderer";

import Container from "./Container";

describe("Container component", () => {
  it("should render without throwing an error", () => {
    const tree = renderer.create(<Container />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render child element", () => {
    const tree = renderer
      .create(
        <Container>
          <p>test</p>
        </Container>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render when having props passed", () => {
    const tree = renderer.create(<Container customStyle="bg-lemon" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
