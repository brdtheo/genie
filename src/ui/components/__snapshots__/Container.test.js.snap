// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should render child element 1`] = `
<div
  className="w-full px-4 lg:px-0 lg:w-[56rem] m-auto "
  data-testid="container"
>
  <p>
    test
  </p>
</div>
`;

exports[`Container component should render when having props passed 1`] = `
<div
  className="w-full px-4 lg:px-0 lg:w-[56rem] m-auto bg-lemon"
  data-testid="container"
/>
`;

exports[`Container component should render without throwing an error 1`] = `
<div
  className="w-full px-4 lg:px-0 lg:w-[56rem] m-auto "
  data-testid="container"
/>
`;
